aten/src/ATen/cuda/tunable/GemmCublasLt.h

```diff
@@ -0,0 +1,595 @@
+}
+
+template <typename T>
+const void* GetBScalePointerFromParams(const GemmStridedBatchedParams<T>* params) {
+  return nullptr;
+}
+
+template <typename T>
+const void* GetBScalePointerFromParams(const ScaledGemmParams<T>* params) {
+  return params->b_scale_ptr;
+}
+
+template <typename T>
+const void* GetDScalePointerFromParams(const GemmParams<T>* params) {
+  return nullptr;
+}
+
+template <typename T>
+const void* GetDScalePointerFromParams(const GemmStridedBatchedParams<T>* params) {
+  return nullptr;
+}
+
+template <typename T>
+const void* GetDScalePointerFromParams(const ScaledGemmParams<T>* params) {
+  return params->c_scale_ptr;
+}
+
+template <typename T>
+const void* GetAmaxPointerFromParams(const GemmParams<T>* params) {
+  return nullptr;
+}
+
+template <typename T>
+const void* GetAmaxPointerFromParams(const GemmStridedBatchedParams<T>* params) {
+  return nullptr;
+}
+
+template <typename T>
+const void* GetAmaxPointerFromParams(const ScaledGemmParams<T>* params) {
+  return params->amax_ptr;
+}
+
+template <typename T>
+int8_t GetFastAccumModeFromParams(const GemmParams<T>* params) {
+  return 0;
+}
+
+template <typename T>
+int8_t GetFastAccumModeFromParams(const GemmStridedBatchedParams<T>* params) {
+  return 0;
+}
+
+template <typename T>
+int8_t GetFastAccumModeFromParams(const ScaledGemmParams<T>* params) {
+  return params->use_fast_accum ? 1 : 0;
+}
+
+static cublasOperation_t MapLayoutToCuBlasLt(BlasOp layout) {
+    if (layout == BlasOp::N) {
+        return CUBLAS_OP_N;
+    }
+    return CUBLAS_OP_T;
+}
+
+template <typename T>
+const void* GetBiasPointerFromParams(const GemmParams<T>*params) {
+    return nullptr;
+}
+
+template <typename T>
+const void* GetBiasPointerFromParams(const GemmStridedBatchedParams<T>* params) {
+  return nullptr;
+}
+
+template <typename T>
+const void* GetBiasPointerFromParams(const ScaledGemmParams<T>* params) {
+  return params->bias_ptr;
+}
+
+template <typename T>
+cudaDataType_t GetBiasTypeFromParams(const GemmParams<T>* params) {
+    return CUDA_R_32F;
+}
+
+template <typename T>
+cudaDataType_t GetBiasTypeFromParams(const GemmStridedBatchedParams<T>* params) {
+    return CUDA_R_32F;
+}
+
+template <typename T>
+cudaDataType_t GetBiasTypeFromParams(const ScaledGemmParams<T>* params) {
+  return at::cuda::ScalarTypeToCudaDataType(params->bias_dtype);
+}
+
+static char _charFromcublasOp(cublasOperation_t op) {
+    switch (op) {
+        case CUBLAS_OP_N:
+            return 'N';
+        case CUBLAS_OP_T:
+            return 'T';
+        case CUBLAS_OP_C:
+            return 'C';
+        default:
+        AT_ERROR("_charFromcublasOp input should be CUBLAS_OP_N/T/C but got '", op, "'");
+    }
+}
+
+template <typename T, BlasOp ALayout, BlasOp BLayout, typename ParamsT>
+class CublasltGemmOp : public Callable<ParamsT> {
+    public:
+        CublasltGemmOp(cublasLtMatmulAlgo_t algo) : algo_{algo} {}
+
+        TuningStatus Call(const ParamsT* params) override {         
```

Comment:
Is some of the code here duplicated with the non-tuned cuBLASLt call? (not saying we need to deduplicate, but might be worth leaving a comment pointing out that the tunable/non-tunable paths should stay in-sync) if that's the case